adafruit_emc2101/__init__.py

```diff
@@ -232,30 +195,77 @@ def initialize(self):
         self._spin_tach_limit = False
         self._calculate_full_speed()
 
+    @property
+    def part_info(self):
+        """The part information: manufacturer, part id and revision.
+        Normally returns (0x5d, 0x16, 0x1).
+        """
+        return (self._mfg_id, self._part_id, self._part_rev)
+
+    @property
+    def devconfig(self):
+        """Read the main device config register.
+        See the CONFIG_* bit definitions in the emc2101_regs module, or refer
+        to the datasheet for more detail. Note: this is not the Fan Config
+        register!
+        """
+        return self._config
+
+    @property
+    def devstatus(self):
+        """Read device status (alerts) register. See the STATUS_* bit
+        definitions in the emc2101_regs module, or refer to the datasheet for
+        more detail.
+        """
+        return self._status
+
     @property
     def internal_temperature(self):
-        """The temperature as measured by the EMC2101's internal 8-bit temperature sensor"""
+        """The temperature as measured by the EMC2101's internal 8-bit
+        temperature sensor, which validly ranges from 0 to 85 and does not
+        support fractions (unlike the external readings).
+
+        :return: int temperature in degrees centigrade.
+        """
         return self._int_temp
 
     @property
     def external_temperature(self):
-        """The temperature measured using the external diode"""
+        """The temperature measured using the external diode. The value is
+        read as a fixed-point 11-bit value ranging from -64 to approx 126,
+        with fractional part of 1/8 degree.
+
+        :return: float temperature in degrees centigrade.
+
+        :raises RuntimeError: if the sensor pind (DP,DN) are open circuit
+            (the sensor is disconnected).
+        :raises RuntimeError: if the external temp sensor is a short circuit
+            (not behaving like a diode).
+        """
 
         temp_lsb = self._ext_temp_lsb
         temp_msb = self._ext_temp_msb
         full_tmp = (temp_msb << 8) | temp_lsb
         full_tmp >>= 5
-        full_tmp *= 0.125
+        if full_tmp == emc2101_regs.TEMP_FAULT_OPENCIRCUIT:
+            raise RuntimeError("Open circuit")
+        if full_tmp == emc2101_regs.TEMP_FAULT_SHORT:
+            raise RuntimeError("Short circuit")
 
-        return full_tmp
+        full_tmp *= 0.125
+        return float(full_tmp)
```

Comment:
This use of `float()` can be removed since the previous line guarantees it's already `float`

Reply:
done